pizza_cutter/slice_utils/interpolate.py

```diff
@@ -74,7 +77,138 @@ def _interp_patch(*, image, bad_msk, i, j, size, buff):
         return None
 
 
+@njit
+def _get_nearby_good_pixels(image, bad_msk, buff=4):
+    """
+    get the set of good pixels surrounding bad pixels.
+
+    Parameters
+    ----------
+    image: array
+        The image data
+    bad_msk: bool array
+        2d array of mask bits.  True means it is a bad
+        pixel
+
+    Returns
+    -------
+    bad_pix:
+        bad pix is the set of bad pixels, shape [nbad, 2]
+    good_pix:
+        good pix is the set of bood pixels around the bad
+        pixels, shape [ngood, 2]
+    good_im:
+        the set of good image values, shape [ngood]
+    good_ind:
+        the 1d indices of the good pixels row*ncol + col
+    """
+
+    nrows, ncols = bad_msk.shape
+
+    ngood = bad_msk.size*3
+    good_pix = np.zeros((ngood, 2), dtype=numba.int64)
+    good_ind = np.zeros(ngood, dtype=numba.int64)
+    bad_pix = np.zeros((ngood, 2), dtype=numba.int64)
+    good_im = np.zeros(ngood, dtype=image.dtype)
+
+    ibad = 0
+    igood = 0
+    for row in range(nrows):
+        for col in range(ncols):
+            val = bad_msk[row, col]
+            if val:
+                bad_pix[ibad] = (row, col)
+                ibad += 1
+
+                row_start = row - buff
+                row_end = row + buff
+                col_start = col - buff
+                col_end = col + buff
+
+                if row_start < 0:
+                    row_start = 0
+                if row_end > (nrows-1):
+                    row_end = nrows-1
+                if col_start < 0:
+                    col_start = 0
+                if col_end > (ncols-1):
+                    col_end = ncols-1
+
+                for rc in range(row_start, row_end+1):
+                    if rc == row:
+                        continue
+                    for cc in range(col_start, col_end+1):
+                        if cc == col:
+                            continue
+
+                        tval = bad_msk[rc, cc]
+                        if not tval:
```

Comment:
I think we only need this test and not the row and column skips above. I think those are actually excluding pixels in the row and column of the bad pixel.

Reply:
yep